projects/knora-ui/src/lib/viewer/values/time-value/time-value.component.ts

```diff
@@ -0,0 +1,198 @@
+import { Component, OnInit, OnChanges, OnDestroy, ViewChild, Input, Inject, SimpleChanges, NgZone } from '@angular/core';
+import { CdkTextareaAutosize } from '@angular/cdk/text-field';
+import { take } from 'rxjs/operators';
+import { TimeInputComponent, DateTime } from './time-input/time-input.component';
+import { ReadTimeValue, CreateTimeValue, UpdateTimeValue, KnoraDate } from '@knora/api';
+import { BaseValueComponent } from '..';
+import { FormControl, FormGroup, FormBuilder, ValidatorFn, AbstractControl } from '@angular/forms';
+import { Subscription } from 'rxjs';
+import { IntervalErrorStateMatcher } from '../interval-value/interval-value.component';
+import { DatePipe } from '@angular/common';
+import * as _ from 'lodash';
+import { CustomRegex } from '../custom-regex';
+
+@Component({
+  selector: 'kui-time-value',
+  templateUrl: './time-value.component.html',
+  styleUrls: ['./time-value.component.scss']
+})
+export class TimeValueComponent extends BaseValueComponent implements OnInit, OnChanges, OnDestroy {
+
+  @ViewChild('timeInput', {static: false}) timeInputComponent: TimeInputComponent;
+  @ViewChild('autosize', {static: false}) autosize: CdkTextareaAutosize;
+
+  @Input() displayValue?: ReadTimeValue;
+
+  valueFormControl: FormControl;
+  commentFormControl: FormControl;
+  date: string;
+  time: string;
+  initKnoraDate: KnoraDate; // used to display the initial value which will be an instance of KnoraDate
+  convertedControlDate: KnoraDate; // used to check the validity of the initial KnoraDate
+  updateDate: Date; // used in order to easily convert a Javascript Date object into a timestamp string using .toISOString()
+
+  form: FormGroup;
+
+  valueChangesSubscription: Subscription;
+
+  customValidators = [];
+
+  matcher = new IntervalErrorStateMatcher();
+
+  constructor(@Inject(FormBuilder) private fb: FormBuilder, private _ngZone: NgZone) {
+    super();
+  }
+
+  standardValidatorFunc: (val: any, comment: string, commentCtrl: FormControl) => ValidatorFn
+    = (initValue: any, initComment: string, commentFormControl: FormControl): ValidatorFn => {
+    return (control: AbstractControl): { [key: string]: any } | null => {
+      
+      if(control.value.date !== null && !(control.value.date instanceof KnoraDate)){
+        this.convertedControlDate = new KnoraDate("Gregorian",
+                                                  "AD",
+                                                  control.value.date.calendarStart.year,
+                                                  control.value.date.calendarStart.month,
+                                                  control.value.date.calendarStart.day);
+      }
+
+      const invalid = (control.value !== null &&
+                       (_.isEqual(initValue.date, this.convertedControlDate) || initValue.date === control.value.date) &&
```

Comment:
`initValue.date === control.value.date`: I think checking for equality makes only sense for primitive values, not for objects. We do not care if it is still the same object, we just care if its value is equal to what we initially had.

Reply:
Removed in 046010e